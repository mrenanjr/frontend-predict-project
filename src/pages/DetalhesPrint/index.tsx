/* eslint-disable jsx-a11y/alt-text */
import React from 'react';
import { AlunoDetail } from '../Detalhes';
import { Document, Page, Text, View, Image, StyleSheet } from '@react-pdf/renderer';
import ImgSchool from '../../assets/outline_school_black_18dp.png'; 

const styles = StyleSheet.create({
    whiteContainer: {
        width: '100%',
        height: '100%',
        marginBottom: 30,
        padding: 30,
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
        flexDirection: 'column',
    },
    header: {
        width: '100%',
        display: 'flex',
        justifyContent: 'space-between',
        alignItems: 'flex-start',
        flexDirection: 'row'
    },
    firstHeader: {
        backgroundColor: 'gray',
        height: '10vh',
        width: '70%',
        paddingLeft: 10,
        display: 'flex',
        flexDirection: 'column',
        justifyContent: 'center',
        alignItems: 'flex-start',
        color: 'black',
        fontSize: 12
    },
    secondHeader: {
        backgroundColor: 'gray',
        height: '10vh',
        width: '28%',
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
        flexDirection: 'column',
        fontSize: 12
    },
    detailsprinttitle: {
        width: '100%',
        marginTop: 10,
        marginBottom: 10,
        fontSize: 18
    },
    img: {
        width: 24
    },
    subtitle: {
        fontSize: 14,
        width: '100%'
    },
    devider: {
        width: '100%',
        marginTop: 10,
        borderBottom: '3 solid orange'
    },
    dados: {
        width: '100%',
        flexDirection: 'row',
        justifyContent: 'space-between'
    },
    elementbox: {
        fontSize: 10,
        width: '30%',
        padding: '5 5 5 0'
    },
    elementboxfive: {
        fontSize: 10,
        width: '20%',
        padding: '5 5 5 0'
    },
    elementboxfirst: {
        backgroundColor: 'gray',
        color: 'white',
        padding: 5,
        borderBottom: '2px solid white'
    },
    elementboxsecond: {
        backgroundColor: 'lightcyan',
        color: 'black',
        padding: 5
    },
    footer: {
        fontSize: 12,
        width: '100%',
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'flex-start'
    }
})

interface DetailObJ {
    matricula: string,
    aluno: AlunoDetail
};

const DetalhesPrint = ({ matricula, aluno }: DetailObJ) => {
    return (
        <Document>
            <Page size="A4" style={styles.whiteContainer}>
                <View style={styles.header}>
                    <View style={styles.firstHeader}>
                        <Text>NOME DA IES</Text>
                        <Text>Endereço, Cidade, Estado e CEP</Text>
                        <Text>Telefone telefone E-mail Redes Sociais</Text>
                    </View>
                    <View style={styles.secondHeader}>
                        <Text>LOGO TIPO DA IES</Text>
                        <Image src={ImgSchool} style={styles.img}/>
                    </View>
                </View>
                <Text style={styles.detailsprinttitle}>RELATÓRIO INDIVIDUAL DA PREDIÇÃO DE EVASÃO</Text>
                <Text style={styles.subtitle}>DADOS DO DISCENTE</Text>
                <View style={styles.devider}></View>
                <View style={styles.dados}>
                    <View style={styles.elementbox}>
                        <Text style={styles.elementboxfirst}>MATRÍCULA</Text>
                        <Text style={styles.elementboxsecond}>{matricula}</Text>
                    </View>
                    <View style={styles.elementbox}>
                        <Text style={styles.elementboxfirst}>NOME DO CURSO</Text>
                        <Text style={styles.elementboxsecond}>{aluno.curso}</Text>
                    </View>
                    <View style={styles.elementbox}>
                        <Text style={styles.elementboxfirst}>PROBABILIDADE DE EVASÃO (%)</Text>
                        <Text style={styles.elementboxsecond}>{aluno.probabilidade_evasao}%</Text>
                    </View>
                </View>
                <View style={styles.dados}>
                    <View style={styles.elementbox}>
                        <Text style={styles.elementboxfirst}>FORMA DE INGRESSO</Text>
                        <Text style={styles.elementboxsecond}>{aluno.forma_ingresso}</Text>
                    </View>
                    <View style={styles.elementbox}>
                        <Text style={styles.elementboxfirst}>ESCOLA ENSINO MÉDIO</Text>
                        <Text style={styles.elementboxsecond}>{aluno.escola_ensino_medio}</Text>
                    </View>
                    <View style={styles.elementbox}>
                        <Text style={styles.elementboxfirst}>NOTA DO IDEB*</Text>
                        <Text style={styles.elementboxsecond}>{aluno.ideb_escola_ensino_medio}</Text>
                    </View>
                </View>
                <Text style={styles.subtitle}>DADOS SOCIO-ECONÔMICOS</Text>
                <View style={styles.devider}></View>
                <View style={styles.dados}>
                    <View style={styles.elementboxfive}>
                        <Text style={styles.elementboxfirst}>ANO DE INGRESSO</Text>
                        <Text style={styles.elementboxsecond}>{aluno.ano_ingresso}</Text>
                    </View>
                    <View style={styles.elementboxfive}>
                        <Text style={styles.elementboxfirst}>CIDADE</Text>
                        <Text style={styles.elementboxsecond}>{aluno.cidade_endereco}</Text>
                    </View>
                    <View style={styles.elementboxfive}>
                        <Text style={styles.elementboxfirst}>ETNIA</Text>
                        <Text style={styles.elementboxsecond}>{aluno.cor_raca}</Text>
                    </View>
                    <View style={styles.elementboxfive}>
                        <Text style={styles.elementboxfirst}>ESCOLA PÚBLICA</Text>
                        <Text style={styles.elementboxsecond}>{aluno.escola_publica}</Text>
                    </View>
                    <View style={styles.elementboxfive}>
                        <Text style={styles.elementboxfirst}>IDADE INGRESSO</Text>
                        <Text style={styles.elementboxsecond}>{aluno.idade_ingresso}</Text>
                    </View>
                </View>
                <Text style={styles.subtitle}>DADOS ACADÊMICOS</Text>
                <View style={styles.devider}></View>
                <View style={styles.dados}>
                    <View style={styles.elementboxfive}>
                        <Text style={styles.elementboxfirst}>MODALIDADE DO CURSO</Text>
                        <Text style={styles.elementboxsecond}>{aluno.modalidade}</Text>
                    </View>
                    <View style={styles.elementboxfive}>
                        <Text style={styles.elementboxfirst}>TURNO</Text>
                        <Text style={styles.elementboxsecond}>{aluno.turno}</Text>
                    </View>
                    <View style={styles.elementboxfive}>
                        <Text style={styles.elementboxfirst}>TRANCAMENTO</Text>
                        <Text style={styles.elementboxsecond}>{aluno.total_trancamentos}</Text>
                    </View>
                    <View style={styles.elementboxfive}>
                        <Text style={styles.elementboxfirst}>MGA**</Text>
                        <Text style={styles.elementboxsecond}>{aluno.media_global_aluno}</Text>
                    </View>
                    <View style={styles.elementboxfive}>
                        <Text style={styles.elementboxfirst}>INTEGRALIZAÇÃO (%)</Text>
                        <Text style={styles.elementboxsecond}>{aluno.percentual_integralizado}%</Text>
                    </View>
                </View>
                <Text>{"\n"}{"\n"}{"\n"}{"\n"}{"\n"}{"\n"}</Text>
                <View style={styles.footer}>
                     <Text>LEGENDA:</Text>
                     <Text>* IDEB - Índice de Desenvolvimento da Educação Básica</Text>
                     <Text>** MGA - Média Global do Aluno</Text>
                 </View>
            </Page>
        </Document>
    );
}

export default DetalhesPrint;