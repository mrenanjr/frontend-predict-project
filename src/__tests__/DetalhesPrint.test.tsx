import * as React from 'react';
import { shallow } from 'enzyme';
import DetalhesPrint from '../pages/DetalhesPrint';

const TestDetalhesPrintValues = {
    matricula: '201300766',
    alunoDetail: {
        ano_ingresso: 2013,
        categoria_ingresso: '',
        cidade_endereco: 'Goiania',
        cor_raca: 'Não quis declarar cor/raça',
        curso: 'CIÊNCIA DA COMPUTAÇÃO',
        deficiencia: '',
        escola_ensino_medio: 'COLÉGIO EST.JOÃO TAVARES MARTINS',
        escola_publica: 'NÃO',
        especificidade_ingresso: '',
        forma_ingresso: 'INGRESSO POR TRANSFERÊNCIA',
        grau_academico: 'BACHARELADO',
        idade_ingresso: 20,
        ideb_escola_ensino_medio: 'Sem cadastro no Censo da Educação Básica 2017',
        media_global_aluno: 3.35,
        media_global_curso: 4.97,
        modalidade: 'Presencial',
        percentual_integralizado: 59.57,
        probabilidade_evasao: 60.2,
        semestre_ingresso: 1,
        sexo: 'MASCULINO',
        status: 'sim',
        total_trancamentos: 0,
        turno: 'Integral',
        uf_endereco: 'GO',
        usuario: 'Administrador'
    }
}

test('DetalhesPrint loads corretly', () => {
  const wrapper = shallow(<DetalhesPrint matricula={TestDetalhesPrintValues.matricula} aluno={TestDetalhesPrintValues.alunoDetail} />);

  expect(wrapper).toMatchSnapshot();
});