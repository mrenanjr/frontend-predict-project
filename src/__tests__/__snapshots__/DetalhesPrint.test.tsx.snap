// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetalhesPrint loads corretly 1`] = `
<DOCUMENT>
  <PAGE
    size="A4"
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
        "height": "100%",
        "justifyContent": "center",
        "marginBottom": 30,
        "padding": 30,
        "width": "100%",
      }
    }
  >
    <VIEW
      style={
        Object {
          "alignItems": "flex-start",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "width": "100%",
        }
      }
    >
      <VIEW
        style={
          Object {
            "alignItems": "flex-start",
            "backgroundColor": "gray",
            "color": "black",
            "display": "flex",
            "flexDirection": "column",
            "fontSize": 12,
            "height": "10vh",
            "justifyContent": "center",
            "paddingLeft": 10,
            "width": "70%",
          }
        }
      >
        <TEXT>
          NOME DA IES
        </TEXT>
        <TEXT>
          Endereço, Cidade, Estado e CEP
        </TEXT>
        <TEXT>
          Telefone telefone E-mail Redes Sociais
        </TEXT>
      </VIEW>
      <VIEW
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "gray",
            "display": "flex",
            "flexDirection": "column",
            "fontSize": 12,
            "height": "10vh",
            "justifyContent": "center",
            "width": "28%",
          }
        }
      >
        <TEXT>
          LOGO TIPO DA IES
        </TEXT>
        <IMAGE
          src="test-file-stub"
          style={
            Object {
              "width": 24,
            }
          }
        />
      </VIEW>
    </VIEW>
    <TEXT
      style={
        Object {
          "fontSize": 18,
          "marginBottom": 10,
          "marginTop": 10,
          "width": "100%",
        }
      }
    >
      RELATÓRIO INDIVIDUAL DA PREDIÇÃO DE EVASÃO
    </TEXT>
    <TEXT
      style={
        Object {
          "fontSize": 14,
          "width": "100%",
        }
      }
    >
      DADOS DO DISCENTE
    </TEXT>
    <VIEW
      style={
        Object {
          "borderBottom": "3 solid orange",
          "marginTop": 10,
          "width": "100%",
        }
      }
    />
    <VIEW
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "width": "100%",
        }
      }
    >
      <VIEW
        style={
          Object {
            "fontSize": 10,
            "padding": "5 5 5 0",
            "width": "30%",
          }
        }
      >
        <TEXT
          style={
            Object {
              "backgroundColor": "gray",
              "borderBottom": "2px solid white",
              "color": "white",
              "padding": 5,
            }
          }
        >
          MATRÍCULA
        </TEXT>
        <TEXT
          style={
            Object {
              "backgroundColor": "lightcyan",
              "color": "black",
              "padding": 5,
            }
          }
        >
          201300766
        </TEXT>
      </VIEW>
      <VIEW
        style={
          Object {
            "fontSize": 10,
            "padding": "5 5 5 0",
            "width": "30%",
          }
        }
      >
        <TEXT
          style={
            Object {
              "backgroundColor": "gray",
              "borderBottom": "2px solid white",
              "color": "white",
              "padding": 5,
            }
          }
        >
          NOME DO CURSO
        </TEXT>
        <TEXT
          style={
            Object {
              "backgroundColor": "lightcyan",
              "color": "black",
              "padding": 5,
            }
          }
        >
          CIÊNCIA DA COMPUTAÇÃO
        </TEXT>
      </VIEW>
      <VIEW
        style={
          Object {
            "fontSize": 10,
            "padding": "5 5 5 0",
            "width": "30%",
          }
        }
      >
        <TEXT
          style={
            Object {
              "backgroundColor": "gray",
              "borderBottom": "2px solid white",
              "color": "white",
              "padding": 5,
            }
          }
        >
          PROBABILIDADE DE EVASÃO (%)
        </TEXT>
        <TEXT
          style={
            Object {
              "backgroundColor": "lightcyan",
              "color": "black",
              "padding": 5,
            }
          }
        >
          60.2
          %
        </TEXT>
      </VIEW>
    </VIEW>
    <VIEW
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "width": "100%",
        }
      }
    >
      <VIEW
        style={
          Object {
            "fontSize": 10,
            "padding": "5 5 5 0",
            "width": "30%",
          }
        }
      >
        <TEXT
          style={
            Object {
              "backgroundColor": "gray",
              "borderBottom": "2px solid white",
              "color": "white",
              "padding": 5,
            }
          }
        >
          FORMA DE INGRESSO
        </TEXT>
        <TEXT
          style={
            Object {
              "backgroundColor": "lightcyan",
              "color": "black",
              "padding": 5,
            }
          }
        >
          INGRESSO POR TRANSFERÊNCIA
        </TEXT>
      </VIEW>
      <VIEW
        style={
          Object {
            "fontSize": 10,
            "padding": "5 5 5 0",
            "width": "30%",
          }
        }
      >
        <TEXT
          style={
            Object {
              "backgroundColor": "gray",
              "borderBottom": "2px solid white",
              "color": "white",
              "padding": 5,
            }
          }
        >
          ESCOLA ENSINO MÉDIO
        </TEXT>
        <TEXT
          style={
            Object {
              "backgroundColor": "lightcyan",
              "color": "black",
              "padding": 5,
            }
          }
        >
          COLÉGIO EST.JOÃO TAVARES MARTINS
        </TEXT>
      </VIEW>
      <VIEW
        style={
          Object {
            "fontSize": 10,
            "padding": "5 5 5 0",
            "width": "30%",
          }
        }
      >
        <TEXT
          style={
            Object {
              "backgroundColor": "gray",
              "borderBottom": "2px solid white",
              "color": "white",
              "padding": 5,
            }
          }
        >
          NOTA DO IDEB*
        </TEXT>
        <TEXT
          style={
            Object {
              "backgroundColor": "lightcyan",
              "color": "black",
              "padding": 5,
            }
          }
        >
          Sem cadastro no Censo da Educação Básica 2017
        </TEXT>
      </VIEW>
    </VIEW>
    <TEXT
      style={
        Object {
          "fontSize": 14,
          "width": "100%",
        }
      }
    >
      DADOS SOCIO-ECONÔMICOS
    </TEXT>
    <VIEW
      style={
        Object {
          "borderBottom": "3 solid orange",
          "marginTop": 10,
          "width": "100%",
        }
      }
    />
    <VIEW
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "width": "100%",
        }
      }
    >
      <VIEW
        style={
          Object {
            "fontSize": 10,
            "padding": "5 5 5 0",
            "width": "20%",
          }
        }
      >
        <TEXT
          style={
            Object {
              "backgroundColor": "gray",
              "borderBottom": "2px solid white",
              "color": "white",
              "padding": 5,
            }
          }
        >
          ANO DE INGRESSO
        </TEXT>
        <TEXT
          style={
            Object {
              "backgroundColor": "lightcyan",
              "color": "black",
              "padding": 5,
            }
          }
        >
          2013
        </TEXT>
      </VIEW>
      <VIEW
        style={
          Object {
            "fontSize": 10,
            "padding": "5 5 5 0",
            "width": "20%",
          }
        }
      >
        <TEXT
          style={
            Object {
              "backgroundColor": "gray",
              "borderBottom": "2px solid white",
              "color": "white",
              "padding": 5,
            }
          }
        >
          CIDADE
        </TEXT>
        <TEXT
          style={
            Object {
              "backgroundColor": "lightcyan",
              "color": "black",
              "padding": 5,
            }
          }
        >
          Goiania
        </TEXT>
      </VIEW>
      <VIEW
        style={
          Object {
            "fontSize": 10,
            "padding": "5 5 5 0",
            "width": "20%",
          }
        }
      >
        <TEXT
          style={
            Object {
              "backgroundColor": "gray",
              "borderBottom": "2px solid white",
              "color": "white",
              "padding": 5,
            }
          }
        >
          ETNIA
        </TEXT>
        <TEXT
          style={
            Object {
              "backgroundColor": "lightcyan",
              "color": "black",
              "padding": 5,
            }
          }
        >
          Não quis declarar cor/raça
        </TEXT>
      </VIEW>
      <VIEW
        style={
          Object {
            "fontSize": 10,
            "padding": "5 5 5 0",
            "width": "20%",
          }
        }
      >
        <TEXT
          style={
            Object {
              "backgroundColor": "gray",
              "borderBottom": "2px solid white",
              "color": "white",
              "padding": 5,
            }
          }
        >
          ESCOLA PÚBLICA
        </TEXT>
        <TEXT
          style={
            Object {
              "backgroundColor": "lightcyan",
              "color": "black",
              "padding": 5,
            }
          }
        >
          NÃO
        </TEXT>
      </VIEW>
      <VIEW
        style={
          Object {
            "fontSize": 10,
            "padding": "5 5 5 0",
            "width": "20%",
          }
        }
      >
        <TEXT
          style={
            Object {
              "backgroundColor": "gray",
              "borderBottom": "2px solid white",
              "color": "white",
              "padding": 5,
            }
          }
        >
          IDADE INGRESSO
        </TEXT>
        <TEXT
          style={
            Object {
              "backgroundColor": "lightcyan",
              "color": "black",
              "padding": 5,
            }
          }
        >
          20
        </TEXT>
      </VIEW>
    </VIEW>
    <TEXT
      style={
        Object {
          "fontSize": 14,
          "width": "100%",
        }
      }
    >
      DADOS ACADÊMICOS
    </TEXT>
    <VIEW
      style={
        Object {
          "borderBottom": "3 solid orange",
          "marginTop": 10,
          "width": "100%",
        }
      }
    />
    <VIEW
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "width": "100%",
        }
      }
    >
      <VIEW
        style={
          Object {
            "fontSize": 10,
            "padding": "5 5 5 0",
            "width": "20%",
          }
        }
      >
        <TEXT
          style={
            Object {
              "backgroundColor": "gray",
              "borderBottom": "2px solid white",
              "color": "white",
              "padding": 5,
            }
          }
        >
          MODALIDADE DO CURSO
        </TEXT>
        <TEXT
          style={
            Object {
              "backgroundColor": "lightcyan",
              "color": "black",
              "padding": 5,
            }
          }
        >
          Presencial
        </TEXT>
      </VIEW>
      <VIEW
        style={
          Object {
            "fontSize": 10,
            "padding": "5 5 5 0",
            "width": "20%",
          }
        }
      >
        <TEXT
          style={
            Object {
              "backgroundColor": "gray",
              "borderBottom": "2px solid white",
              "color": "white",
              "padding": 5,
            }
          }
        >
          TURNO
        </TEXT>
        <TEXT
          style={
            Object {
              "backgroundColor": "lightcyan",
              "color": "black",
              "padding": 5,
            }
          }
        >
          Integral
        </TEXT>
      </VIEW>
      <VIEW
        style={
          Object {
            "fontSize": 10,
            "padding": "5 5 5 0",
            "width": "20%",
          }
        }
      >
        <TEXT
          style={
            Object {
              "backgroundColor": "gray",
              "borderBottom": "2px solid white",
              "color": "white",
              "padding": 5,
            }
          }
        >
          TRANCAMENTO
        </TEXT>
        <TEXT
          style={
            Object {
              "backgroundColor": "lightcyan",
              "color": "black",
              "padding": 5,
            }
          }
        >
          0
        </TEXT>
      </VIEW>
      <VIEW
        style={
          Object {
            "fontSize": 10,
            "padding": "5 5 5 0",
            "width": "20%",
          }
        }
      >
        <TEXT
          style={
            Object {
              "backgroundColor": "gray",
              "borderBottom": "2px solid white",
              "color": "white",
              "padding": 5,
            }
          }
        >
          MGA**
        </TEXT>
        <TEXT
          style={
            Object {
              "backgroundColor": "lightcyan",
              "color": "black",
              "padding": 5,
            }
          }
        >
          3.35
        </TEXT>
      </VIEW>
      <VIEW
        style={
          Object {
            "fontSize": 10,
            "padding": "5 5 5 0",
            "width": "20%",
          }
        }
      >
        <TEXT
          style={
            Object {
              "backgroundColor": "gray",
              "borderBottom": "2px solid white",
              "color": "white",
              "padding": 5,
            }
          }
        >
          INTEGRALIZAÇÃO (%)
        </TEXT>
        <TEXT
          style={
            Object {
              "backgroundColor": "lightcyan",
              "color": "black",
              "padding": 5,
            }
          }
        >
          59.57
          %
        </TEXT>
      </VIEW>
    </VIEW>
    <TEXT>
      

      

      

      

      

      

    </TEXT>
    <VIEW
      style={
        Object {
          "alignItems": "flex-start",
          "display": "flex",
          "flexDirection": "column",
          "fontSize": 12,
          "width": "100%",
        }
      }
    >
      <TEXT>
        LEGENDA:
      </TEXT>
      <TEXT>
        * IDEB - Índice de Desenvolvimento da Educação Básica
      </TEXT>
      <TEXT>
        ** MGA - Média Global do Aluno
      </TEXT>
    </VIEW>
  </PAGE>
</DOCUMENT>
`;
